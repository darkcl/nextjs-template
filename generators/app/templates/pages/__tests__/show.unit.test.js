import React from 'react';
import toJson from 'enzyme-to-json';
import { render } from 'enzyme';

import ApiClient from '../../lib/api-client';
import Show from '../show';

const apiClient = new ApiClient('.');

describe('<Show />', () => {
  it('getInitialProps should return object with status code', async () => {
    let props = await Show.getInitialProps({ query: { id: '975' } });

    expect(props).toBeDefined();
    expect(props).toBeDefined();
    expect(props.statusCode).toBe(200);
    expect(props.data.id).toBe(975);

    props = await Show.getInitialProps({ query: { id: '123456789' } });

    expect(props).toBeDefined();
    expect(props).toBeDefined();
    expect(props).toBeDefined();
    expect(props.statusCode).toBe(404);
  });

  it('should render correctly with searchable result', async () => {
    const res = await apiClient.getShow('975');

    const show = render(<Show {...res} />);
    expect(toJson(show)).toMatchSnapshot();
  });

  it('should render correctly with non-searchable result', async () => {
    const res = await apiClient.getShow('123456789');

    const show = render(<Show {...res} />);
    expect(toJson(show)).toMatchSnapshot();
  });
});
