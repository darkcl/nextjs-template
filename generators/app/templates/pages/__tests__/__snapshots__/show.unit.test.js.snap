// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Show /> should render correctly with non-searchable result 1`] = `
Array [
  <div
    class="header"
  >
    <div
      class="container"
    >
      <a
        href="/"
      >
        <img
          src="undefined/static/47cd76e43f74bbc2e1baaf194d07e1fa/images/favicon.png"
        />
      </a>
      <span>
        header.title
      </span>
    </div>
  </div>,
  <div
    class="container"
  >
    <p>
      An error 404 occurred on server
    </p>
  </div>,
  <div
    class="footer"
  >
    <div
      class="container"
    >
      footer.text
    </div>
  </div>,
]
`;

exports[`<Show /> should render correctly with searchable result 1`] = `
Array [
  <div
    class="header"
  >
    <div
      class="container"
    >
      <a
        href="/"
      >
        <img
          src="undefined/static/47cd76e43f74bbc2e1baaf194d07e1fa/images/favicon.png"
        />
      </a>
      <span>
        header.title
      </span>
    </div>
  </div>,
  <div
    class="container"
  >
    <div
      class="show"
    >
      <h1>
        Batman
      </h1>
      <div
        class="row"
      >
        <div
          class="col-md-3"
        >
          <img
            class="img-fluid"
            src="http://static.tvmaze.com/uploads/images/medium_portrait/6/16463.jpg"
          />
        </div>
        <div
          class="col-md-9"
        >
          <div>
            <p>
              Wealthy entrepreneur Bruce Wayne and his ward Dick Grayson lead a double life: they are actually crime fighting duo Batman and Robin. A secret Batpole in the Wayne mansion leads to the Batcave, where Police Commissioner Gordon often calls with the latest emergency threatening Gotham City. Racing to the scene of the crime in the Batmobile, Batman and Robin must (with the help of their trusty Bat-utility-belt) thwart the efforts of a variety of master criminals, including Catwoman, Egghead, The Joker, King Tut, The Penguin, and The Riddler.
            </p>
          </div>
        </div>
      </div>
      <div
        class="row text-center"
      >
        <div
          class="col-md-12"
        >
          <a
            class="btn btn-primary"
            href="/"
          >
            Back
          </a>
        </div>
      </div>
    </div>
  </div>,
  <div
    class="footer"
  >
    <div
      class="container"
    >
      footer.text
    </div>
  </div>,
]
`;
